langchain/llms/base.py

```diff
@@ -48,7 +49,12 @@ def generate(
                 raise ValueError(
                     "Asked to cache, but no cache found at `langchain.cache`."
                 )
-            return self._generate(prompts, stop=stop)
+            get_tracer().start_llm_trace({"name": self.__class__.__name__}, prompts)
+            output = self._generate(prompts, stop=stop)
+            get_tracer().end_llm_trace(
```

Comment:
output is of type `LLMResult`... I know we talked about the base tracer taking in `LLMResult` as well, but that would result in a circular import. Even placing `LLMResult` in a separate file within `llms` would cause this problem bc of the imports within `__init__.py`

Reply:
im down to put it in https://github.com/hwchase17/langchain/blob/master/langchain/schema.py, Generation is already there